sdk/core/AzureCore/Source/DataStructures/CancellationToken.swift

```diff
@@ -39,6 +43,7 @@ public final class CancellationToken: Codable, Equatable {
         self.timeoutInSeconds = timeoutInSeconds
         self.isStarted = false
         self.isCanceled = false
+        self.runId = Int.random(in: Int.min ... Int.max)
     }
 
     // MARK: Equatable Protocol
```

Comment:
We should replace this implementation with one that just compares the runIDs since they are supposed to be unique